src/conda_package_handling/conda_fmt.py

```diff
@@ -47,47 +32,69 @@ def supported(fn):
     @staticmethod
     def extract(fn, dest_dir, **kw):
         components = utils.ensure_list(kw.get("components")) or ("info", "pkg")
-        file_id = os.path.basename(fn).replace(".conda", "")
         if not os.path.isabs(fn):
             fn = os.path.normpath(os.path.join(os.getcwd(), fn))
         if not os.path.isdir(dest_dir):
             os.makedirs(dest_dir)
-        for component in components:
-            _extract_component(fn, file_id, component, dest_dir)
+
+        _extract(str(fn), str(dest_dir), components=components)
 
     @staticmethod
     def extract_info(fn, dest_dir=None):
         return CondaFormat_v2.extract(fn, dest_dir, components=["info"])
 
     @staticmethod
-    def create(prefix, file_list, out_fn, out_folder=os.getcwd(), **kw):
+    def create(
+        prefix,
+        file_list,
+        out_fn,
+        out_folder=os.getcwd(),
+        compressor=lambda: zstandard.ZstdCompressor(
```

Comment:
Could we remove this inline definition? I have to admit it looks really weird to see a lambda function as the default value for a staticmethod. You can just define it as a regular function above this class and provide that as the default.

Also, you could at least use the `Callable` type hint to let others know that this is a callable object. Otherwise, it could be a little confusing at first glance.

Reply:
I am a little surprised that you are unaccustomed to the lambda syntax.